django/db/models/deletion.py

```diff
@@ -251,11 +299,33 @@ def collect(self, objs, source=None, nullable=False, collect_related=True,
             for batch in batches:
                 sub_objs = self.related_objects(related_model, related_fields, batch)
                 self.fast_deletes.append(sub_objs)
+                # self.add_dependency(model, related_model) Do we need this?
```

Comment:
This change looks unrelated. I commented this line and I don't see any failures.

Reply:
Agree this is unrelated and removed this line